packages/reactive-element/src/reactive-element.ts

```diff
@@ -668,6 +668,12 @@ export abstract class ReactiveElement
     (this.__controllers ??= []).push(controller);
   }
 
+  removeController(controller: ReactiveController) {
+    if (this.__controllers !== undefined) {
+      this.__controllers = this.__controllers?.filter((c) => c !== controller);
+    }
```

Comment:
Check this out:

```suggestion
    this.__controllers?.splice(this.__controllers?.indexOf(c) >>> 0, 1);
```

Usually we should expect that add/remove calls are balanced, so `indexOf(c)` will have the correct index to remove. If for some reason `c` isn't in the list, `indexOf(c)` will return `-1`, and `splice()` would remove the last item, but `-1 >>> 0` strips the sign bit and returns `0xFFFFFFFF`. `splice()` doesn't remove anything if the first arg is greater than the length of the array, so that's a no op.

Reply:
On the current version you can also remove the optional chaining.

Reply:
Nifty, changed.